contracts/src/periphery/EigenDACertVerifierV1.sol

```diff
@@ -0,0 +1,152 @@
+     * @param blobHeaders Pointer to array of blob headers in calldata
+     * @param blobVerificationProofs Pointer to array of blob cert verification proofs in calldata
+     */
+    function verifyDACertsV1(BlobHeader[] calldata blobHeaders, BlobVerificationProof[] calldata blobVerificationProofs)
+        external
+        view
+    {
+        // Check length match
+        if (blobHeaders.length != blobVerificationProofs.length) {
+            revert LengthMismatch();
+        }
+
+        // Verify each blob
+        for (uint256 i; i < blobHeaders.length; ++i) {
+            verifyDACertV1(blobHeaders[i], blobVerificationProofs[i]);
+        }
+    }
+
+    /**
+     * @notice Verifies a blob cert and returns result without reverting
+     * @param blobHeader Pointer to the blob header in calldata
+     * @param blobVerificationProof Pointer to the blob cert verification proof in calldata
+     * @return success True if verification succeeded
+     */
+    function verifyDACertV1ForZkProof(
+        BlobHeader calldata blobHeader,
+        BlobVerificationProof calldata blobVerificationProof
+    ) external view returns (bool success) {
+        (CertLib.ErrorCode errorCode,) = CertLib.verifyDACert(
+            _quorumConfirmationThresholdPercentages(),
+            _storedBatchMetadataHash(blobVerificationProof),
+            blobHeader,
+            blobVerificationProof,
+            _quorumNumbersRequired()
+        );
+        if (errorCode == CertLib.ErrorCode.SUCCESS) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @notice Returns the threshold registry contract
+     * @return The IEigenDAThresholdRegistry contract
+     * @dev Can be overridden by derived contracts
+     */
+    function _thresholdRegistry() internal view virtual returns (IEigenDAThresholdRegistry) {
+        return eigenDAThresholdRegistryV1;
+    }
+
+    /**
+     * @notice Returns the batch metadata storage contract
+     * @return The IEigenDABatchMetadataStorage contract
+     * @dev Can be overridden by derived contracts
+     */
+    function _batchMetadataStorage() internal view virtual returns (IEigenDABatchMetadataStorage) {
+        return eigenDABatchMetadataStorageV1;
+    }
+
+    /**
+     * @notice Returns the quorum numbers required for verification
+     * @return bytes The required quorum numbers
+     * @dev Can be overridden by derived contracts
+     */
+    function _quorumNumbersRequired() internal view virtual returns (bytes memory) {
+        return _thresholdRegistry().quorumNumbersRequired();
+    }
+
+    /**
+     * @notice Returns the quorum confirmation threshold percentages
+     * @return bytes The confirmation threshold percentages for each quorum
+     * @dev Can be overridden by derived contracts
+     */
+    function _quorumConfirmationThresholdPercentages() internal view virtual returns (bytes memory) {
```

Comment:
@samlaf #1472 

> does bytes behave the same as []uint8 in solidity? If so we can change the type

The behavior is actually quite different, but I am a fan of changing the type anyway. However, this is a threshold registry level change, so should probably be part of another PR.

Reply:
Wait so a uint8 array actually uses 32 bytes per uint8??

Reply:
in memory, yes.